### PR TITLE
Find invoices in batches

### PR DESCRIPTION
**Pull request by SorchaAbel**

### CODE REVIEW

File: lib/xeroizer/record/base_model.rb [hunk 1]
```diff
@@ -122,6 +122,17 @@ def all(options = {})
           response.response_items || []
         end
 
+        # allow invoices to be process in batches of 100 as per xero documentation
+        # https://developer.xero.com/documentation/api/invoices/
+        def find_in_batches(options = {}, &block)
+          i = 1
+          while results = all(page: i)
```
Comment (CloCkWeRX) [inline lib/xeroizer/record/base_model.rb]: @SorchaAbel can you merge options with the page counter above

File: lib/xeroizer/record/base_model.rb [hunk 2]
```diff
@@ -122,6 +122,17 @@ def all(options = {})
           response.response_items || []
         end
 
+        # allow invoices to be process in batches of 100 as per xero documentation
+        # https://developer.xero.com/documentation/api/invoices/
+        def find_in_batches(options = {}, &block)
+          i = options[:page_number]
```
Comment (nnc) [inline lib/xeroizer/record/base_model.rb]: You should probably default to 1 if `:page_number` option was not set.

Comment (SorchaAbel) [inline lib/xeroizer/record/base_model.rb]: Thanks @nnc

**Comment by CloCkWeRX** [comment]
I think this one might be ready to go

**Comment by waynerobinson** [comment]
It looks mostly fine, but is there a reason why it uses `block.call()` instead of a straight `yield`? This was a potential performance issue (although I doubt it's that relevant in the grand scheme of things.

**Comment by nnc** [comment]
Any reason not to pass `options` to `all` method in `find_in_batches`? I see that `first` method does that, and that would also make `find_in_batches` much more versatile IMO.

**Comment by CloCkWeRX** [comment]
So... I'm both happy with this in general and sitting next to @SorchaAbel right now; but want to make sure there's consensus before whacking the merge button - are the latest batch of changes good enough @nnc

**Comment by waynerobinson** [comment]
This seems fine to me too.

**Comment by nnc** [comment]
Yes, looks great. Will put it to use now. Thanks @SorchaAbel.

**Pull request merged**